typeql/reasoner/negation.feature

```diff
@@ -1202,3 +1202,71 @@ Feature: Negation Resolution
       """
     Then verify answer size is: 0
 #    TODO: Add a case with non-zero answers
+
+
+  Scenario: Negated relation which is both retrievable and concludable must always consider both (issue#6500)
```

Comment:
Good title, but reading it I realised that we actually don't use `retrievable` and `concludable` as terminology in the behaviour tests (the tests largely predate this terminology). This is probably best as it disassociates the implementation from the test scenarios. We refer to `inferred relation` a lot, I'm yet to find the terminology we should use for the non-inferred case. Please take a look or improvise.